test/conformance/serial/ptp.go

```diff
@@ -838,6 +840,225 @@ var _ = Describe("["+strings.ToLower(DesiredMode.String())+"-serial]", Serial, f
+						openshift_ptp_process_status{config="ts2phc.0.config",node="cnfde22.ptp.lab.eng.bos.redhat.com",process="gpsd"} 1
+						openshift_ptp_process_status{config="ts2phc.0.config",node="cnfde22.ptp.lab.eng.bos.redhat.com",process="gpspipe"}
+						openshift_ptp_process_status{config="ts2phc.0.config",node="cnfde22.ptp.lab.eng.bos.redhat.com",process="gpspipe"} 1
+						openshift_ptp_process_status{config="ts2phc.0.config",node="cnfde22.ptp.lab.eng.bos.redhat.com",process="ts2phc"} 1
+					*/
+					Eventually(func() string {
+						buf, _, _ := pods.ExecCommand(client.Client, fullConfig.DiscoveredClockUnderTestPod, pkg.PtpContainerName, []string{"curl", pkg.MetricsEndPoint})
+						return buf.String()
+					}, pkg.TimeoutIn5Minutes, 5*time.Second).Should(ContainSubstring(metrics.OpenshiftPtpProcessStatus),
+						"Process status metrics are not detected")
+
+					Eventually(func() string {
+						buf, _, _ := pods.ExecCommand(client.Client, fullConfig.DiscoveredClockUnderTestPod, pkg.PtpContainerName, []string{"curl", pkg.MetricsEndPoint})
+						return buf.String()
+					}, pkg.TimeoutIn5Minutes, 5*time.Second).Should(ContainSubstring("phc2sys"),
+						"phc2ys process status not detected")
+
+					time.Sleep(10 * time.Second)
+					buf, _, _ := pods.ExecCommand(client.Client, fullConfig.DiscoveredClockUnderTestPod, pkg.PtpContainerName, []string{"curl", pkg.MetricsEndPoint})
+					ret := processRunning(buf.String())
+					Expect(ret["phc2sys"]).To(BeTrue(), "Expected phc2sys to be running for GM")
+					Expect(ret["gpspipe"]).To(BeTrue(), "Expected gpspipe to be running for GM")
+					Expect(ret["ts2phc"]).To(BeTrue(), "Expected ts2phc to be running for GM")
+					Expect(ret["gpsd"]).To(BeTrue(), "Expected gpsd to be running for GM")
+					Expect(ret["ptp4l"]).To(BeTrue(), "Expected ptp4l to be running for GM")
+
+				})
+
+				By("checking clock class state is locked", func() {
+					/*
+						# TYPE openshift_ptp_clock_class gauge
+						# openshift_ptp_clock_class{node="cnfdg32.ptp.eng.rdu2.dc.redhat.com",process="ptp4l"} 6
+					*/
+					Eventually(func() string {
+						buf, _, _ := pods.ExecCommand(client.Client, fullConfig.DiscoveredClockUnderTestPod, pkg.PtpContainerName, []string{"curl", pkg.MetricsEndPoint})
+						return buf.String()
+					}, pkg.TimeoutIn5Minutes, 5*time.Second).Should(ContainSubstring(metrics.OpenshiftPtpClockClass),
+						"NMEA status metrics are not detected")
+					buf, _, _ := pods.ExecCommand(client.Client, fullConfig.DiscoveredClockUnderTestPod, pkg.PtpContainerName, []string{"curl", pkg.MetricsEndPoint})
+					clockClassPattern := `openshift_ptp_clock_class\{node="([^"]+)",process="([^"]+)"\} (\d+)`
+
+					// Compile the regular expression
+					clockClassRe := regexp.MustCompile(clockClassPattern)
+
+					// Find matches
+					clockClassMap := map[string]bool{"ptp4l": false}
+
+					scanner := bufio.NewScanner(strings.NewReader(buf.String()))
+					for scanner.Scan() {
```

Comment:
should have a timeout here for the for loop

Reply:
I have added 1 minute time out for the two methods I created to parse the metrics